### DISCUSSION TITLE
[QUESTION]: How to get cvc5 value in python base api without crash?

### DISCUSSION THREAD

**Discussion by hagozaebii** [cvc5/cvc5#10477] [Q&A]
hello

I have been using multiprocessing pipes to avoid crashes that occur in the cvc5 base API. Specifically, we used a method of performing the calculation of the cvc5 API through multiprocessing and returning the calculated value if it is executed within time. However, when using this method, the Python primitive type obtained with `toPythonObj()` is passed through a pipe, but the disadvantage is that it cannot be obtained if the cvc5 value itself is needed. This is because cvc5 instances are not serialized/deserialized through pickle.

Is there a good way to get the value of cvc5 while avoiding a crash?

**Comment by srod5125** [comment]
Maybe getValue, getProof, or getModel in solver methods.

**Reply by hagozaebii** [reply]
getValue makes crash.

**Comment by daniel-larraz** [comment]
@hagozaebii Can you please provide a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) that causes a crash?
